tests/integration/error_warning_file_tests.py

```diff
@@ -0,0 +1,521 @@
+import os
+import csv
+
+from dataactcore.interfaces.db import GlobalDB
+from dataactcore.config import CONFIG_SERVICES
+from dataactcore.models.jobModels import Submission
+from dataactcore.models.domainModels import concat_tas_dict
+from tests.unit.dataactcore.factories.domain import SF133Factory, TASFactory
+from dataactvalidator.health_check import create_app
+from dataactvalidator.validation_handlers.validationManager import ValidationManager
+from dataactcore.models.lookups import (FILE_TYPE_DICT, JOB_TYPE_DICT, JOB_STATUS_DICT)
+from dataactbroker.handlers.fileHandler import report_file_name
+
+from tests.integration.baseTestValidator import BaseTestValidator
+from tests.integration.integration_test_helper import insert_submission, insert_job
+
+FILES_DIR = os.path.join('tests', 'integration', 'data')
+
+# Valid Files
+APPROP_FILE_T = os.path.join(FILES_DIR, 'appropValid.csv')
+CROSS_FILE_A = os.path.join(FILES_DIR, 'cross_file_A.csv')
+CROSS_FILE_B = os.path.join(FILES_DIR, 'cross_file_B.csv')
+
+# Invalid Files
+HEADER_ERROR = os.path.join(FILES_DIR, 'appropHeaderError.csv')
+READ_ERROR = os.path.join(FILES_DIR, 'appropReadError.csv')
+LENGTH_ERROR = os.path.join(FILES_DIR, 'appropLengthError.csv')
+TYPE_ERROR = os.path.join(FILES_DIR, 'appropTypeError.csv')
+REQUIRED_ERROR = os.path.join(FILES_DIR, 'appropRequiredError.csv')
+RULE_FAILED_WARNING = os.path.join(FILES_DIR, 'appropInvalidWarning.csv')
+RULE_FAILED_ERROR = os.path.join(FILES_DIR, 'appropInvalidError.csv')
+INVALID_CROSS_A = os.path.join(FILES_DIR, 'invalid_cross_file_A.csv')
+INVALID_CROSS_B = os.path.join(FILES_DIR, 'invalid_cross_file_B.csv')
+
+
+class ErrorWarningTests(BaseTestValidator):
+    """
+    Overall integration tests for error/warning reports.
+
+    For each file type (single-file, cross-file, errors, warnings), test if each has
+        - the correct structure
+        - each column's content is correct after testing each possible type of error:
+            - formatting
+            - length
+            - types
+            - required/optional
+            - SQL validation
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """ Set up class-wide resources (test data) """
+        super(ErrorWarningTests, cls).setUpClass()
+
+        with create_app().app_context():
+            # get the submission test users
+            sess = GlobalDB.db().session
+            cls.session = sess
+            cls.admin_user_id = 1
```

Comment:
I know we only have one right now, but this is bad practice if we ever need multiple users. We should be getting the user and storing their ID instead